src/native/libs/System.Native/pal_time.c

```diff
@@ -53,6 +53,33 @@ int32_t SystemNative_UTimensat(const char* path, TimeSpec* times)
     return result;
 }
 
+int32_t SystemNative_FUTimens(intptr_t fd, TimeSpec* times)
+{
+    int32_t result;
+
+#if HAVE_FUTIMENS
+    struct timespec updatedTimes[2];
+    updatedTimes[0].tv_sec = (time_t)times[0].tv_sec;
+    updatedTimes[0].tv_nsec = (long)times[0].tv_nsec;
+    updatedTimes[1].tv_sec = (time_t)times[1].tv_sec;
+    updatedTimes[1].tv_nsec = (long)times[1].tv_nsec;
+
+    while (CheckInterrupted(result = futimens(ToFileDescriptor(fd), updatedTimes)));
+#else
```

Comment:
this should probably be `#else if HAVE_FUTIMES` since per the code above apparently we have to handle the case where we don't have it. (Seems like Linux has had it since 2016, so not sure it's relevant but we should be consistent)

If we don't even have `futimes` then we should return failure from this function.

Reply:
@tmds Can you confirm that? Or is this some special case?

Reply:
With these two cases we cover all CI platforms.

If a platform gets added that doesn't support `futimens` and `futimes`, changes will need to be made.

I have a personal preference for that being a compile time error (as it is now) because it is very visible.